mlflow/types/schema.py

```diff
@@ -870,13 +870,21 @@ def numpy_types(self) -> List[np.dtype]:
         """Convenience shortcut to get the datatypes as numpy types."""
         if self.is_tensor_spec():
             return [x.type for x in self.inputs]
-        return [x.type.to_numpy() for x in self.inputs]
+        if all(isinstance(x.type, DataType) for x in self.inputs):
+            return [x.type.to_numpy() for x in self.inputs]
+        raise MlflowException(
+            "Failed to get numpy types as some of the inputs types are not DataType."
```

Comment:
Can we make this error message a little more clear. Do users know what `DataType` is? That's more of an implementation detail. Can we tell the users precisely what went wrong if their input caused this exception to be raised (and how to fix it)?

Reply:
I think this function is not called anywhere. It used to be called for schema check, but I removed it because now we have more types that are not DataType.

Reply:
The only 2 references to it are for 2 tests in a suite. I guess theoretically it's a public utility function?
Although how anyone would find this outside of really digging through the API docs or the source code is beyond me. https://www.mlflow.org/docs/latest/python_api/mlflow.types.html#mlflow.types.Schema.numpy_types

Reply:
Yea that's true. If users call it I think they know what datatypes are. And if they still just use old ColSpecs with DataTypes, it keeps the same behavior.